cocotb/share/lib/gpi/GpiCommon.cpp

```diff
@@ -160,18 +161,34 @@ static void gpi_load_libs(std::vector<std::string> to_load)
          iter++)
     {
         void *lib_handle = NULL;
-        std::string full_name = "lib" + *iter + DOT_LIB_EXT;
+
+        std::string arg = *iter;
+
+        std::string lib_name;
+        std::string func_name;
+        auto it = std::find(arg.begin(), arg.end(), ':');
+        if (it == arg.end()) {
+            // no colon in the string, default
+            lib_name = arg;
+            func_name = arg + "_entry_point";
+        }
+        else {
+            lib_name = std::string(arg.begin(), it);
+            func_name = std::string(it+1, arg.end());
+        }
+
+        std::string full_name = "lib" + lib_name + DOT_LIB_EXT;
         const char *now_loading = (full_name).c_str();
 
         lib_handle = utils_dyn_open(now_loading);
         if (!lib_handle) {
             printf("cocotb: Error loading shared library %s\n", now_loading);
             exit(1);
         }
-        std::string sym = (*iter) + "_entry_point";
-        void *entry_point = utils_dyn_sym(lib_handle, sym.c_str());
+
+        void *entry_point = utils_dyn_sym(lib_handle, func_name.c_str());
         if (!entry_point) {
-            printf("cocotb: Unable to find entry point %s for shared library %s\n", sym.c_str(), now_loading);
+            printf("cocotb: Unable to find entry point %s for shared library %s\n", func_name.c_str(), now_loading);
```

Comment:
I think we want to tell users why things changed here, just in case:
```suggestion
            char const* fmt = "cocotb: Unable to find entry point %s for shared library %s\n%s";
            char const* msg = "        Perhaps you meant to use `,` instead of `:` to separate library names, as this changed in cocotb 1.4?\n";
            printf(fmt, func_name.c_str(), full_name.c_str(), has_colon ? msg : "");
```